docs/logging_snippets.py

```diff
@@ -0,0 +1,369 @@
+    UPDATED_FILTER = 'textPayload:robot'
+    UPDATED_DESCRIPTION = "Danger, Will Robinson!"
+
+    # [START client_list_metrics]
+    for metric in client.list_metrics():  # API call(s)
+        do_something_with(metric)
+    # [END client_list_metrics]
+
+    # [START metric_create]
+    metric = client.metric(
+        METRIC_NAME, filter_=FILTER, description=DESCRIPTION)
+    assert not metric.exists()  # API call
+    metric.create()  # API call
+    assert metric.exists()  # API call
+    # [END metric_create]
+    to_delete.append(metric)
+
+    # [START metric_reload]
+    existing_metric = client.metric(METRIC_NAME)
+    existing_metric.reload()  # API call
+    # [END metric_reload]
+    assert existing_metric.filter_ == FILTER
+    assert existing_metric.description == DESCRIPTION
+
+    # [START metric_update]
+    existing_metric.filter_ = UPDATED_FILTER
+    existing_metric.description = UPDATED_DESCRIPTION
+    existing_metric.update()  # API call
+    # [END metric_update]
+    existing_metric.reload()
+    assert existing_metric.filter_ == UPDATED_FILTER
+    assert existing_metric.description == UPDATED_DESCRIPTION
+
+    def _metric_delete():
+        # [START metric_delete]
+        metric.delete()
+        # [END metric_delete]
+
+    _backoff_not_found(_metric_delete)
+    to_delete.remove(metric)
+
+
+def _sink_storage_setup(client):
+    from google.cloud import storage
+    BUCKET_NAME = 'sink-storage-%d' % (_millis(),)
+    client = storage.Client()
+    bucket = client.bucket(BUCKET_NAME)
+    bucket.create()
+
+    # [START sink_bucket_permissions]
+    bucket.acl.reload()
+    logs_group = bucket.acl.group('cloud-logs@google.com')
+    logs_group.grant_owner()
+    bucket.acl.add_entity(logs_group)
+    bucket.acl.save()
+    # [END sink_bucket_permissions]
+
+    return bucket
+
+
+@snippet
+def sink_storage(client, to_delete):
+    """Sink log entries to storage."""
+    bucket = _sink_storage_setup(client)
+    to_delete.append(bucket)
+    SINK_NAME = 'robots-storage-%d' % (_millis(),)
+    FILTER = 'textPayload:robot'
+
+    # [START sink_storage_create]
+    DESTINATION = 'storage.googleapis.com/%s' % (bucket.name,)
+    sink = client.sink(SINK_NAME, filter_=FILTER, destination=DESTINATION)
+    assert not sink.exists()  # API call
+    sink.create()  # API call
+    assert sink.exists()  # API call
+    # [END sink_storage_create]
+    to_delete.insert(0, sink)  # delete sink before bucket
+
+
+def _sink_bigquery_setup(client):
+    from google.cloud import bigquery
+    DATASET_NAME = 'sink_bigquery_%d' % (_millis(),)
+    client = bigquery.Client()
+    dataset = client.dataset(DATASET_NAME)
+    dataset.create()
+    dataset.reload()
+
+    # [START sink_dataset_permissions]
+    from google.cloud.bigquery.dataset import AccessGrant
+    grants = dataset.access_grants
+    grants.append(AccessGrant(
+        'WRITER', 'groupByEmail', 'cloud-logs@google.com'))
```

Comment:
1. Create the grant on it's on and then append?
1. Why not append to `dataset.access_grants` in place? (Is it a `@property` that returns a copy?)